### DISCUSSION TITLE
homematic IP Geräte lassen sich nicht über Home Assistant Addon anlernen

### DISCUSSION THREAD

**Discussion by Marc-Roth** [OpenCCU/OpenCCU#2420] [🇩🇪 Fragen&Antworten]
### Describe the issue you are experiencing

Es können keine Geräte über das Home Assistant Addon angelernt werden.

### Describe the behavior you expected

Befindet sich ein homematic IP Gerät  im Anlernmodus, sollte über die Funktion _HmIP Gerät anlernen_, des RaspberryMatic WebUI, eine Verbindung hergestellt werden.

### Steps to reproduce the issue

1. Home Assistant Operating System installieren und aktualisieren
2. Home Assistant Addon installieren
3. mit CCU3 gekoppeltes homematic IP Gerät auf Werkseinstellungen zurücksetzen (hier HmIP-eTRV-2)
4. CCU3 vom Netz nehmen
5. Anlernmodus auf dem homematic IP Gerät aktivieren
6. Anlernmodus über die WebUI aktivieren

### What is the version this bug report is based on?

3.71.12.20230826

### Which base platform are you running?

rpi3 (RaspberryPi3)

### Which HomeMatic/homematicIP radio module are you using?

HmIP-RFUSB

### Anything in the logs that might be useful for us?

```txt
-
```

### Additional information

# Hardware

Raspberry Pi 3b
Sonoff Zigbee 3.0 USB Dongle
Silicon Labs eQ-3 HmIP-RFUSB

# Software

Home Assistant 2023.8.4
Supervisor 2023.08.1
Operating System 10.5
Frontend 20230802.1 - latest

# Logs

##  dmesg | grep usb

```
[    0.160957] usbcore: registered new interface driver usbfs
[    0.161068] usbcore: registered new interface driver hub
[    0.161196] usbcore: registered new device driver usb
[    0.161776] usb_phy_generic phy: supply vcc not found, using dummy regulator
[    0.162139] usb_phy_generic phy: dummy supplies not allowed for exclusive requests
[    2.417117] dwc_otg 3f980000.usb: DWC OTG Controller
[    2.418717] dwc_otg 3f980000.usb: new USB bus registered, assigned bus number 1
[    2.420364] dwc_otg 3f980000.usb: irq 74, io mem 0x00000000
[    2.425842] usb usb1: New USB device found, idVendor=1d6b, idProduct=0002, bcdDevice= 6.01
[    2.429120] usb usb1: New USB device strings: Mfr=3, Product=2, SerialNumber=1
[    2.430864] usb usb1: Product: DWC OTG Controller
[    2.432600] usb usb1: Manufacturer: Linux 6.1.21-v8 dwc_otg_hcd
[    2.434345] usb usb1: SerialNumber: 3f980000.usb
[    2.443689] usbcore: registered new interface driver usb-storage
[    2.445578] usbcore: registered new interface driver usbserial_generic
[    2.447312] usbserial: USB Serial support registered for generic
[    2.470667] usbcore: registered new interface driver usbhid
[    2.471608] usbhid: USB HID core driver
[    2.721430] usb 1-1: new high-speed USB device number 2 using dwc_otg
[    2.911795] usb 1-1: New USB device found, idVendor=0424, idProduct=9514, bcdDevice= 2.00
[    2.914605] usb 1-1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    3.206359] usb 1-1.1: new high-speed USB device number 3 using dwc_otg
[    3.299719] usb 1-1.1: New USB device found, idVendor=0424, idProduct=ec00, bcdDevice= 2.00
[    3.299743] usb 1-1.1: New USB device strings: Mfr=0, Product=0, SerialNumber=0
[    3.379391] usb 1-1.2: new full-speed USB device number 4 using dwc_otg
[    3.477361] usb 1-1.2: New USB device found, idVendor=10c4, idProduct=ea60, bcdDevice= 1.00
[    3.480763] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    3.482517] usb 1-1.2: Product: Sonoff Zigbee 3.0 USB Dongle Plus
[    3.484161] usb 1-1.2: Manufacturer: Silicon Labs
[    3.485797] usb 1-1.2: SerialNumber: 0001
[    3.568336] usb 1-1.5: new full-speed USB device number 5 using dwc_otg
[    3.667359] usb 1-1.5: New USB device found, idVendor=1b1f, idProduct=c020, bcdDevice= 1.00
[    3.669482] usb 1-1.5: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[    3.670579] usb 1-1.5: Product: eQ-3 HmIP-RFUSB
[    3.671635] usb 1-1.5: Manufacturer: Silicon Labs
[    3.672668] usb 1-1.5: SerialNumber: 3014F711A0000420C9A1826B
[    6.212581] usbcore: registered new interface driver cp210x
[    6.213972] usbserial: USB Serial support registered for cp210x
[    6.245529] usb 1-1.2: cp210x converter now attached to ttyUSB0
[    6.342959] usb 1-1.5: Found eQ-3 HmIP-RFUSB with serial 3014F711A0000420C9A1826B at usb-3f980000.usb-1.5
[    6.350782] raw-uart raw-uart: Registered new raw-uart device using underlying device eQ-3 HmIP-RFUSB@usb-3f980000.usb-1.5.
[    6.355988] usbcore: registered new interface driver hb_rf_usb_2
[    6.530692] SMSC LAN8700 usb-001:003:01: attached PHY driver (mii_bus:phy_addr=usb-001:003:01, irq=184)
[    6.568940] smsc95xx 1-1.1:1.0 eth0: register 'smsc95xx' at usb-3f980000.usb-1.1, smsc95xx USB 2.0 Ethernet, b8:27:eb:24:48:17
[    6.579450] usbcore: registered new interface driver smsc95xx
[    6.862640] usbcore: registered new interface driver brcmfmac
```

## RaspberryMatic CCU Protokoll

```
Identifying Homematic RF-Hardware: ....HmRF: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-3f980000.usb-1.5, HmIP: HMIP-RFUSB/eQ-3 HmIP-RFUSB@usb-3f980000.usb-1.5, OK
Updating Homematic RF-Hardware: HMIP-RFUSB: 4.4.18, not necessary, OK
...
Starting HMIPServer: .......................OK
```

**Comment by jens-maus** [comment]
Da dies kein genereller Fehler bzw. Bug in RaspberryMatic sein kann/ist, habe ich das mal in den Diskussionsbereich verschoben.

Das Problem bei diesen recht gängigen "Anlern- bzw. Kommunikationsproblemen" mit homematic oder vor allem homematicIP Geräten ist, dass es hierfür mehrere Gründe geben kann. Aus dem Logfile das hier hinzugefügt wurde ist erstmal nicht zu erkennen, dass hier ein Problem existiert. Das Funkmodul (`HmIP-RFUSB`) wurde ganz offensichtlich richtig erkannt und auch wohl richtig eingebunden. Wenn nun auch in der WebUI der Anlernmodus gestartet werden kann und dieser dann von 60s zu 0s runterzählt, ist eigentlich insofern alles bzgl. der Anbindung des Funkmodules an die CCU/RaspberryMatic soweit korrekt.

Das jedoch trotzdem keine HomeMatic/homematicIP Geräte angelernt werden können kann nun natürlich viele Ursachen haben, die jedoch meist in der lokalen Installation/Umgebung zu suchen/finden sind. Die Beliebtesten davon sind:

1. Es existiert ein Störsignal auf der 868MHz Frequenz (verursacht durch einen anderen Aktor/Sensor, Babyphone, Mikrowelle, sonstiges technisches Gerät in der Nähe) oder das Funkmodul bekommt durch die räumliche Nähe zu anderer Technik (z.B. auch das Host system selbst, wie z.B. gängigerweise ein RaspberryPi4) Störsignale ab die ein Anlernen gerade von HmIP Geräten (weil dort listen-before-talk gemacht wird) unmöglich machen. Zu sehen ist dies oft durch einen "CarrierSense"-Wert der dauerhaft oder auch kurzzeitig immer wieder auf > 10% ansteigt. Dieser Wert kann auf der Hauptseite der WebUI eingesehen werden.
2. Die Zentrale ist schon mit anderen Funksignalen ausgelastet und darf daher aus gesetzlichen Gründen aktuell nicht weiter funken: DutyCycle Wert >95% auf Hauptseite der CCU/RaspberryMatic.
3. Die Distanz zwischen Funkmodulantenne und homematic Gerät ist zu groß oder ggf. sogar zu klein.
4. Das HomeMatic/homematicIP Gerät befindet sich nicht korrekt im Anlernmodus.
5. Die CCU/RaspberryMatic Zentrale hat keinen direkten Internetzugriff und kann daher homematicIP Geräte nicht "via Internet" anlernen.
6. Das Funkmodul hat einen Defekt der zwar kein Problem beim Einbinden verursacht, aber die Sende-/Empfangsleistung stark beeinträchtigt und eine Kommunikation zwischen HomeMatic Geräten und dem Funkmodul unmöglich macht.

Um nun ggf. Lösungen für diese lokalen Probleme zu finden bietet sich meistens je nach Ursache folgendes vorgehen an (vgl. Auflistung mit oben:

1. Abstand zwischen Funkmodul und Hostsystem schaffen (z.B. 2-3m USB Verlängerungskabel), Entstörfilter einsetzen (USB Entstörfilter, etc.), Externe Antenne anlöten und diese weit vom Hostsystem absetzen, anderes Hostsystem nutzen (RaspberryPi4 sind Funkstörer!), usw.
2. Alle WebUI Programme, Automatisierungen, etc. abschalten und >1h warten bis DutyCycle wieder < 20% ist. Dann Anlernversuch erneut durchführen. Wenn erfolg, Programme/Automatisierungen ändern das weniger gefunkt wird (Schlagwort: Funkhygiene).
3. Externe Antenne an Funkmodul anbinden (SMA-Pigtail mit professioneller Antennelösung wie Groundplane-Antenne). Funkmodul weiter wegrücken bzw. näher an Ort der HomeMatic/homematicIP Geräte bringen
4. Anleitungen des jeweiligen HomeMatic/homematicIP Gerätes genauestens studieren. Erneuter Werksreset des HomeMatic/homematicIP Gerätes durchführen. Wenn das homematicIP Gerät vorher an der homematicIP Cloud-Lösung (`HmIP-HAP`) angelernt war, sicherstellen das dieses dort komplett abgelernt und frisch werksresettet wurde.
5. Internetzugriff für CCU/RaspberryMatic Zentrale erlauben oder direkten Anlernmodus ohne Internet (eingabe SGTIN+KEY) nutzen.
6. Alle Lötpunkte des Funkmodules erneut nachlöten und vor allem das korrekte anbringen des Funkantennenstummeln kontrollieren oder gleich zur Nutzung einer externen Antennenlösung übergehen um eine bessere Sende-/Empfangsleistung zu haben.

**Comment by Baxxy13** [comment]
Ich spekuliere auf einen weiteren schlecht gelöteten HmIP-RFUSB.

War der schon anderswo erfolgreich im Einsatz?

[>>Hier>>](https://homematic-forum.de/forum/viewtopic.php?f=65&t=76473&hilit=Anlernkatastrophe#p741589) gibt es eine "Analyse-Anleitung" für Anlernprobleme von mir. 
Das mal durchexerzieren und am Ende die nötigen Informationen + Log zeigen.
